src/charm.py

```diff
@@ -55,15 +64,37 @@ def _on_start(self, event: StartEvent):
         Returns:
             None
         """
-        magma_service = subprocess.run(
-            ["systemctl", "is-active", "magma@magmad"],
-            stdout=subprocess.PIPE,
-        )
-        if magma_service.returncode != 0:
+        if not self._magma_service_is_running:
             event.defer()
             return
         self.unit.status = ActiveStatus()
 
+    def _on_get_access_gateway_secrets(self, event: ActionEvent) -> None:
+        """Triggered on get-access-gateway-secrets action call.
+
+        Returns Access Gateway's Hardware ID and Challange Key required to integrate AGW with
+        the Orchestrator.
+        """
+        if not self._magma_service_is_running:
+            event.fail("Magma is not running! Please start Magma and try again.")
+            return
+        try:
+            hardware_id, challenge_key = self._get_magma_secrets
+            logger.error(hardware_id)
```

Comment:
why are you using `error` level?
should this be logged anyway?

Reply:
Already removed.
Just a leftover from testing.

Reply:
I see I commented this the same time you pushed a commit to fix it, do don't worry about it :) @Gmerold